code/modules/martial_arts/cqc.dm

```diff
@@ -39,8 +39,17 @@
 	return ..()
 
 /datum/martial_art/cqc/teach(mob/living/carbon/human/H, make_temporary)
-	var/datum/action/defensive_stance/defensive = new /datum/action/defensive_stance()
-	defensive.Grant(H)
+	var/found = FALSE
+	for(var/datum/martial_art/M in H.mind.known_martial_arts)
+		if(istype(M, /datum/martial_art/cqc/under_siege))
+			M.remove(H)
+	for(var/datum/action/D in H.actions)
+		if(istype(D, /datum/action/defensive_stance))
+			found = TRUE
+			break
```

Comment:
this part isnt needed as that action will be removed by removing the under_siege martial art

Reply:
that's there cos if you use multiple CQC books you get multiple defensive_stance icons